test/QuoterV3.integ.ts

```diff
@@ -0,0 +1,153 @@
+import { expect } from 'chai'
+import { BigNumber } from 'ethers'
+import { QuoterV3 } from '../typechain'
+
+import { JsonRpcSigner } from '@ethersproject/providers'
+import hre, { ethers } from 'hardhat'
+import { encodePath } from './shared/path'
+import { expandTo18Decimals, expandToNDecimals } from './shared/expandTo18Decimals'
+import { FeeAmount, V2_FEE } from './shared/constants'
+
+const V3_FACTORY = '0x1F98431c8aD98523631AE4a59f267346ea31F984'
+const V2_FACTORY = '0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f'
+
+const USDC = '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48'
+const USDT = '0xdAC17F958D2ee523a2206206994597C13D831ec7'
+const WETH = '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2'
+const UNI = '0x1f9840a85d5af5bf1d1762f925bdaddc4201f984'
+const DAI = '0x6B175474E89094C44Da98b954EedeAC495271d0F'
+
+/// @dev basic V2 routes
+const DAI_V2_UNI_V2_WETH = encodePath([DAI, UNI, WETH], [V2_FEE, V2_FEE])
+const USDC_V2_UNI_V2_WETH = encodePath([USDC, UNI, WETH], [V2_FEE, V2_FEE])
+
+/// @dev basic V3 routes
+const USDC_V3_500_USDT = encodePath([USDC, USDT], [FeeAmount.LOW])
+const UNI_V3_3000_WETH = encodePath([UNI, WETH], [FeeAmount.MEDIUM])
+
+/// @dev stablecoin IL routes
+const USDT_V3_500_DAI_V2_USDC = encodePath([USDT, DAI, USDC], [FeeAmount.LOW, V2_FEE])
+const DAI_V3_100_USDC_V2_USDT = encodePath([DAI, USDC, USDT], [100, V2_FEE])
+
+/// @dev erc20 IL routes
+// V3 - V2
+const UNI_V3_3000_WETH_V2_DAI = encodePath([UNI, WETH, DAI], [FeeAmount.MEDIUM, V2_FEE])
+const USDC_V3_3000_UNI_V2_WETH = encodePath([USDC, UNI, WETH], [FeeAmount.MEDIUM, V2_FEE])
+// V2 - V3
+const UNI_V2_WETH_V3_3000_DAI = encodePath([UNI, WETH, DAI], [V2_FEE, FeeAmount.MEDIUM])
+
+/// @dev complex IL routes
+// (use two V3 pools)
+const DAI_V3_3000_UNI_V2_USDT_V3_3000_WETH = encodePath(
```

Comment:
these names are a bit hard to read/hard to know what you're testing. the comments are good but wonder if we could just name them COMPLEX_MIXED_1 = ? Or is it helpful to know the exact path/pools?

Reply:
yeah they are very long. It was very helpful for me when I was piecing routes together from testing, but I can change to a more generic name. 

Additionally, I think we should add a few more integration tests. There aren't that many because they're really hard to validate against SOR (have to piece together route manually). However now that we have more built out it might be easier.